capi/src/run_metadata.rs

```diff
@@ -7,27 +7,29 @@ pub type OwnedRunMetadata = *mut RunMetadata;
 
 #[no_mangle]
 pub unsafe extern "C" fn RunMetadata_run_id(this: *const RunMetadata) -> *const c_char {
-    output_str(acc(this).run_id())
+    output_str(acc(&this).run_id())
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn RunMetadata_platform_name(this: *const RunMetadata) -> *const c_char {
-    output_str(acc(this).platform_name())
+    output_str(acc(&this).platform_name())
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn RunMetadata_uses_emulator(this: *const RunMetadata) -> bool {
-    acc(this).uses_emulator()
+    acc(&this).uses_emulator()
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn RunMetadata_region_name(this: *const RunMetadata) -> *const c_char {
-    output_str(acc(this).region_name())
+    output_str(acc(&this).region_name())
 }
 
+/// # Safety
+/// `this` must outlive `OwnedRunMetadataVariablesIter`
 #[no_mangle]
-pub unsafe extern "C" fn RunMetadata_variables(
+pub unsafe extern "C" fn RunMetadata_variables<'a>(
     this: *const RunMetadata,
-) -> OwnedRunMetadataVariablesIter {
-    alloc(acc(this).variables())
+) -> OwnedRunMetadataVariablesIter<'a> {
```

Comment:
This has no effect whatsoever though. A lifetime always needs to be bound to at least two "things", otherwise the compiler treats it as 'static. So even in this case you don't gain anything from it.

Reply:
However I agree that documenting it is very useful. So I would definitely merge something that just adds some documentation (or maybe the lifetime in these and only these cases).

Reply:
You're right that it doesn't really have an effect, since raw pointers don't carry lifetime information, however this is also about the intent. If someone new looked at this library and saw that lifetime operator, they would be able to see that it must be dependent on some other object, even if the compiler can't enforce that here.

The old `'static` annotations on make me think that some it must reference some static object, which is wrong in this case as it's just a little hack to avoid a couple of lifetime annotations.